libr/asm/p/asm_x86_nz.c

```diff
@@ -1254,13 +1267,12 @@ SETNP/SETPO - Set if No Parity / Set if Parity Odd (386+)
 					return 0;
 				} else {
 					// mov [num], reg
-					ut8 *ptr = (ut8 *)&num;
 					data[l++] = 0x89;
 					data[l++] = (r0 << 3) | 5;
-					data[l++] = ptr[0];
-					data[l++] = ptr[1];
-					data[l++] = ptr[2];
-					data[l++] = ptr[3];
+					data[l++] = num;
```

Comment:
0xFF ensure there's no sign extension in the value which can produce wrong values when higher bit is set. But if it works in that way it should be fine i guess